python/ray/_private/services.py

```diff
@@ -1070,16 +1071,18 @@ def _start_redis_instance(executable,
         command += (["--port", str(port), "--loglevel", "warning"])
         if listen_to_localhost_only:
             command += ["--bind", "127.0.0.1"]
+        fd, pidfile = tempfile.mkstemp()
+        command += ["--pidfile", pidfile]
         process_info = start_ray_process(
             command,
             ray_constants.PROCESS_TYPE_REDIS_SERVER,
             stdout_file=stdout_file,
             stderr_file=stderr_file,
             fate_share=fate_share)
-        time.sleep(0.1)
-        # Check if Redis successfully started (or at least if it the executable
-        # did not exit within 0.1 seconds).
-        if process_info.process.poll() is None:
+        wait_for_redis_to_start("127.0.0.1", port, password=password)
+        r = redis.StrictRedis(host="127.0.0.1", port=port, password=password)
+        # Check if Redis successfully started and we connected to the right server
+        if r.config_get("pidfile")["pidfile"] == pidfile:
```

Comment:
It's great that sleeping is removed. The new approach is more robust and doesn't rely on the guess work of sleep times. It's elegant too.